src/core/context.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Limited
+# See LICENSE file for licensing details.
+
+"""Charm Context definition and parsing logic."""
+
+from charms.data_platform_libs.v0.data_interfaces import DatabaseRequirerData
+from ops import CharmBase, Relation
+
+from constants import (
+    AUTHENTICATION_DATABASE_NAME,
+    METASTORE_DATABASE_NAME,
+    POSTGRESQL_AUTH_DB_REL,
+    POSTGRESQL_METASTORE_DB_REL,
+    S3_INTEGRATOR_REL,
+)
+from core.domain import DatabaseConnectionInfo, S3ConnectionInfo, ServiceAccountInfo
+from utils.logging import WithLogging
+
+
+class Context(WithLogging):
+    """Properties and relations of the charm."""
+
+    def __init__(self, charm: CharmBase):
```

Comment:
I've been checking whether we could use some less wider object, other than charm in the constructor. The reason is that charm is a very wide-scoped object. I would prefer to provide more scoped inputs, like `config` and `model`. 

I believe we can do that as long as the `ServiceAccountInfo` only depends on config. All the other usage in the class would only need `model`, and not `charm`.

Reply:
Updated the `Context` constructor to receive `model` and `config` instead of `charm`.